apps/gdal_translate_lib.cpp

```diff
@@ -2553,6 +2559,31 @@ static void AttachMetadata(GDALDatasetH hDS,
     }
 }
 
+/************************************************************************/
+/*                           AttachDomainMetadata()                           */
+/************************************************************************/
+
+static void AttachDomainMetadata(GDALDatasetH hDS,
+                                 const CPLStringList &aosDomainMetadataOptions)
+
+{
+    const int nCount = aosDomainMetadataOptions.size();
+
+    for (int i = 0; i < nCount; i++)
+    {
+        // FIXME: catch invalid cases syntax is DOMAIN:META-TAG=value
```

Comment:
yes definitely something to do.
I would do the CSLTokenizeString2() on the pszKey, so as to be able to support  DOMAIN:FOO=BAR:WITH:COLUMN
And you should check that aosTokens.size() == 2

Reply:
ah! I see, thanks - should I error out if it's not size 2, or just ignore it?

Reply:
> should I error out if it's not size 2

would make sense to emit a CPLError(CE_Warning, CPLE_IllegalArg, "....");

Reply:
ignore it but with a warning message of an ill-formed argument seems right

Reply:
CPLParseNameValue parses on "=" or ":" so I need to have  closer look

Reply:
> CPLParseNameValue parses on "=" or ":" so I need to have closer look

ah, good point. You'll have to use strchr(str, '=') then to do the splitting.